commands/enrich/enrich.go

```diff
@@ -0,0 +1,304 @@
+	go func() {
+		defer fileProducer.Done()
+		// Iterate over file-spec groups and produce indexing tasks.
+		// When encountering an error, log and move to next group.
+		specFiles := enrichCmd.spec.Files
+		artifactHandlerFunc := enrichCmd.createIndexerHandlerFunc(indexedFileProducer, resultsArr, indexedFileErrors, xrayVersion)
+		taskHandler := getAddTaskToProducerFunc(fileProducer, artifactHandlerFunc)
+
+		err := FileForIndexing(specFiles[0], taskHandler)
+		if err != nil {
+			log.Error(err)
+			fileCollectingErrorsQueue.AddError(err)
+		}
+	}()
+}
+
+func (enrichCmd *EnrichCommand) createIndexerHandlerFunc(indexedFileProducer parallel.Runner, resultsArr [][]*ScanInfo, indexedFileErrors [][]formats.SimpleJsonError, xrayVersion string) FileContext {
+	return func(filePath string) parallel.TaskFunc {
+		return func(threadId int) (err error) {
+			// Add a new task to the second producer/consumer
+			// which will send the indexed binary to Xray and then will store the received result.
+			taskFunc := func(threadId int) (err error) {
+				fileContent, err := os.ReadFile(filePath)
+				if err != nil {
+					return err
+				}
+				params := &services.XrayGraphImportParams{
+					SBOMInput: fileContent,
+					ScanType:  services.Binary,
+				}
+				if enrichCmd.progress != nil {
+					enrichCmd.progress.SetHeadlineMsg("Scanning 🔍")
+				}
+				importGraphParams := enrichgraph.NewEnrichGraphParams().
+					SetServerDetails(enrichCmd.serverDetails).
+					SetXrayGraphScanParams(params).
+					SetXrayVersion(xrayVersion)
+				xrayManager, err := utils.CreateXrayServiceManager(importGraphParams.ServerDetails())
+				if err != nil {
+					return err
+				}
+				scanResults, err := enrichgraph.RunImportGraphAndGetResults(importGraphParams, xrayManager)
+				if err != nil {
+					indexedFileErrors[threadId] = append(indexedFileErrors[threadId], formats.SimpleJsonError{FilePath: filePath, ErrorMessage: err.Error()})
+					return
+				}
+				resultsArr[threadId] = append(resultsArr[threadId], &ScanInfo{Target: filePath, Result: scanResults})
+				return
+			}
+
+			_, _ = indexedFileProducer.AddTask(taskFunc)
+			return
+		}
+	}
+}
+
+func getAddTaskToProducerFunc(producer parallel.Runner, fileHandlerFunc FileContext) indexFileHandlerFunc {
+	return func(filePath string) {
+		taskFunc := fileHandlerFunc(filePath)
+		_, _ = producer.AddTask(taskFunc)
+	}
+}
+
+func (enrichCmd *EnrichCommand) performScanTasks(fileConsumer parallel.Runner, indexedFileConsumer parallel.Runner) {
+	go func() {
+		// Blocking until consuming is finished.
+		fileConsumer.Run()
+		// After all files have been indexed, The second producer notifies that no more tasks will be produced.
+		indexedFileConsumer.Done()
+	}()
+	// Blocking until consuming is finished.
+	indexedFileConsumer.Run()
+}
+
+func FileForIndexing(fileData spec.File, dataHandlerFunc indexFileHandlerFunc) error {
```

Comment:
```suggestion
func FileForScanning(fileData spec.File, dataHandlerFunc indexFileHandlerFunc) error {
```
No indexing in this command...